### PR TITLE
cephfs-journal-tool: enable purge_queue journal's event commands

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Resolves: http://tracker.ceph.com/issues/24604
Signed-off-by: Xuehan Xu <xuxuehan@360.cn>

### CODE REVIEW

**Comment by xxhdx1985126** [comment]
@batrick Hi, thanks for your review:-) I've corrected the code just as you suggested:-)

**Comment by xxhdx1985126** [comment]
retest this, please

**Comment by xxhdx1985126** [comment]
retest this, please

**Comment by xxhdx1985126** [comment]
retest this, please

File: src/mds/PurgeQueue.h [hunk 1]
```diff
@@ -45,11 +50,35 @@ class PurgeItem
...
   PurgeItem()
-   : action(NONE), ino(0), size(0)
+   : stamp(ceph_clock_now()), pad_size(0), action(NONE), ino(0), size(0)
   {}
 
   void encode(bufferlist &bl) const;
   void decode(bufferlist::const_iterator &p);
+
+  static Action str_to_type(std::string_view str) {
+    return PurgeItem::actions.at(std::string(str));
```
Comment (batrick) [inline src/mds/PurgeQueue.h]: Small nit: see for how to lookup using string_view: https://stackoverflow.com/questions/35525777/use-of-string-view-for-map-lookup

**Comment by xxhdx1985126** [comment]
@batrick Hi, thanks for the advice, I've corrected the code as suggested, please take a look:-)

**Comment by batrick** [comment]
Not what I had in mind but using `string_view` keys is safe in this case.

**Comment by ukernel** [comment]
reverted because it breaks backward compatibility

**Comment by xxhdx1985126** [comment]
@ukernel Thanks:-) I got that, going to correct it right away

**Comment by tchaikov** [comment]
@xxhdx1985126 i am not @ukernel and can not present him. i was just guessing what could go wrong w/ your change.

**Comment by xxhdx1985126** [comment]
@tchaikov Oh, sorry. Thanks:-)

**Comment by xxhdx1985126** [comment]
@ukernel @tchaikov I've submitted a new PR with the corresponding correction, please take a look:-)

**Comment by batrick** [comment]
> reverted because it breaks backward compatibility

good catch!

**Pull request merged**